source/common/network/address_impl.cc

```diff
@@ -22,6 +22,35 @@ namespace Address {
 
 namespace {
 
+/**
+ * Convert an IPv4 address to canonical string format.
+ * @note This works similarly to inet_ntop() but is faster.
+ * @param addr address to format.
+ * @return the address in dotted-decimal string format.
+ */
+static std::string Ipv4ToString(const sockaddr_in& addr) {
+  static constexpr size_t BufferSize = 16; // enough space to hold an IPv4 address in string form
```

Comment:
can you add unit-tests for this? I can imagine we should hit some corner cases of 0s and 255s in every octet.

I realize you'd have to make this non-static.

Reply:
+1 on comprehensive UTs. When we override a stdlib function like this I think we need to backfill comprehensive tests that cover all corner cases.